cdc/model/changefeed_test.go

```diff
@@ -334,6 +495,118 @@ func TestFixState(t *testing.T) {
 	}
 }
 
+func TestFixSinkProtocol(t *testing.T) {
+	t.Parallel()
+
+	// Test fixing the protocol in the configuration.
+	configTestCases := []struct {
+		info             *ChangeFeedInfo
+		expectedProtocol config.Protocol
+	}{
+		{
+			info: &ChangeFeedInfo{
+				SinkURI: "mysql://root:test@127.0.0.1:3306/",
+				Config: &config.ReplicaConfig{
+					Sink: &config.SinkConfig{Protocol: config.ProtocolDefault.String()},
+				},
+			},
+			expectedProtocol: config.ProtocolOpen,
+		},
+		{
+			info: &ChangeFeedInfo{
+				SinkURI: "kafka://127.0.0.1:9092/ticdc-test2",
+				Config: &config.ReplicaConfig{
+					Sink: &config.SinkConfig{Protocol: config.ProtocolCanal.String()},
+				},
+			},
+			expectedProtocol: config.ProtocolCanal,
+		},
+		{
+			info: &ChangeFeedInfo{
+				SinkURI: "kafka://127.0.0.1:9092/ticdc-test2",
+				Config: &config.ReplicaConfig{
+					Sink: &config.SinkConfig{Protocol: config.ProtocolDefault.String()},
+				},
+			},
+			expectedProtocol: config.ProtocolOpen,
+		},
+		{
+			info: &ChangeFeedInfo{
+				SinkURI: "kafka://127.0.0.1:9092/ticdc-test2",
+				Config: &config.ReplicaConfig{
+					Sink: &config.SinkConfig{Protocol: "random"},
+				},
+			},
+			expectedProtocol: config.ProtocolOpen,
+		},
+	}
+
+	for _, tc := range configTestCases {
+		tc.info.fixSinkProtocol()
+		var protocol config.Protocol
+		err := protocol.FromString(tc.info.Config.Sink.Protocol)
+		require.Nil(t, err)
+		require.Equal(t, tc.expectedProtocol, protocol)
+	}
+
+	// Test fixing the protocol in SinkURI.
+	sinkURITestCases := []struct {
```

Comment:
What will output if SinkURI is `kafka://127.0.0.1:9092/ticdc-test2?protocol=default` in this case?

Reply:
It will be `kafka://127.0.0.1:9092/ticdc-test2?protocol=open-protocol`. I added a case for it.